app/src/main/java/com/realityexpander/tasky/agenda_feature/data/repositories/syncRepository/syncRepositoryImpls/SyncRepositoryImpl.kt

```diff
@@ -0,0 +1,175 @@
+    }
+
+    override suspend fun syncDeletedAgendaItems(syncItems: List<SyncItemEntity>) : ResultUiText<Void> {
+        val deleteRequest = syncItems.toSyncAgendaRequestDTO()
+
+        // If there are no items to Delete, then there is nothing to sync.
+        if(deleteRequest.deletedEventIds.isNullOrEmpty()
+            && deleteRequest.deletedTaskIds.isNullOrEmpty()
+            && deleteRequest.deletedReminderIds.isNullOrEmpty()
+        ) {
+            return ResultUiText.Success(null)
+        }
+
+        val result = syncApi.syncAgenda(deleteRequest)
+
+        return if (result.isSuccess) {
+            syncItems.forEach {
+                deleteSyncItemByAgendaItemId(it.agendaItemId, it.modificationTypeForSync)
+            }
+            ResultUiText.Success(null)
+        } else {
+            ResultUiText.Error(UiText.Str(result.exceptionOrNull()?.localizedMessage ?: "Unknown error"))
+        }
+    }
+
+    override suspend fun getSyncItems() : List<SyncItemEntity> =
+        syncDao.getSyncAgendaItems()
+
+    //////////////////////////
+    //// HELPERS
+
+    private fun AgendaItem.toAgendaItemTypeForSync(): AgendaItemTypeForSync {
+        return when (this) {
+            is AgendaItem.Task -> AgendaItemTypeForSync.Task
+            is AgendaItem.Event -> AgendaItemTypeForSync.Event
+            is AgendaItem.Reminder -> AgendaItemTypeForSync.Reminder
+            else -> {
+                throw IllegalArgumentException("Unknown AgendaItem type")
+            }
+        }
+    }
+
+    private suspend fun addSyncItem(
+        modificationTypeForSync: ModificationTypeForSync,
+        agendaItemTypeForSync: AgendaItemTypeForSync,
+        agendaItemId: AgendaItemId
+    ): ResultUiText<Void> {
+        return try {
+            syncDao.addSyncEntity(
+                SyncItemEntity(
+                    agendaItemId = agendaItemId,
+                    agendaItemTypeForSync = agendaItemTypeForSync,
+                    modificationTypeForSync = modificationTypeForSync
+                )
+            )
+            ResultUiText.Success(null)
+        } catch (e: Exception) {
+            return ResultUiText.Error(UiText.Str(e.localizedMessage))
+        }
+    }
+
+    override suspend fun deleteSyncItemByAgendaItemId(agendaItemId: AgendaItemId, modificationTypeForSync: ModificationTypeForSync) =
+        syncDao.deleteSyncAgendaItemByAgendaItemId(agendaItemId, modificationTypeForSync)
+
+    // Prepare the OFFLINE-DELETED AgendaItems to be sent to the server
+    private fun List<SyncItemEntity>.toSyncAgendaRequestDTO() =
+        SyncAgendaRequestDTO(
+            deletedEventIds =
+            this
+                .filter {
+                    it.agendaItemTypeForSync == AgendaItemTypeForSync.Event
+                            && it.modificationTypeForSync == ModificationTypeForSync.Deleted
+                }
+                .map {
+                    it.agendaItemId
+                },
+            deletedTaskIds =
```

Comment:
This formatting makes it super hard to read IMO, would make sense here to store each list in a variable before for readability 🤔

Reply:
OK, I will improve the formatting